components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/FxaAccountManagerTest.kt

```diff
@@ -1886,6 +1887,180 @@ class FxaAccountManagerTest {
         verify(syncManager, times(2)).stop()
         verifyNoMoreInteractions(syncManager)
     }
+    @Test
+    fun `beginAuthenticationAuthSync handles any pairingUrl correctly`() {
+        // Initialize things so the test can run
+        WorkManagerTestInitHelper.initializeTestWorkManager(testContext)
+
+        // Build our dependencies for creating an account manager
+        val tServerConfig = ServerConfig(
+                "https://localhost/contentUrl",
+                "testClientId",
+                "https://localhost/redirectUrl"
+        )
+        val tDeviceConfig = DeviceConfig(
+                "testDevice",
+                DeviceType.MOBILE,
+                setOf()
+        )
+        val tSyncConfig = SyncConfig(setOf(SyncEngine.Bookmarks))
+        val manager = FxaAccountManager(
+                testContext,
+                serverConfig = tServerConfig,
+                deviceConfig = tDeviceConfig,
+                syncConfig = tSyncConfig
+        )
+        val pairingUrl = "https://localhost/pairingUrl".toString()
+        val beginResponse = manager.beginAuthenticationAsync(pairingUrl)
+        assertTrue(beginResponse is CompletableDeferred<String?>)
+    }
+
+    @Test
+    fun `beginAuthenticationAsync with null URL handled correctly`() {
+        // Initialize things so the test can run
+        WorkManagerTestInitHelper.initializeTestWorkManager(testContext)
+
+        // Build our dependencies for creating an account manager
+        val tServerConfig = ServerConfig(
+                "https://localhost/contentUrl",
+                "testClientId",
+                "https://localhost/redirectUrl"
+        )
+        val tDeviceConfig = DeviceConfig(
+                "testDevice",
+                DeviceType.MOBILE,
+                setOf()
+        )
+        val tSyncConfig = SyncConfig(setOf(SyncEngine.Bookmarks))
+        val manager = FxaAccountManager(
+                testContext,
+                serverConfig = tServerConfig,
+                deviceConfig = tDeviceConfig,
+                syncConfig = tSyncConfig
+        )
+        val beginResponse = manager.beginAuthenticationAsync(null)
```

Comment:
Same comment as above, currently this is (almost) not testing anything. You want to at least `await` here, and assert the actual returned value.